src/docs/release/breaking-changes/actions-api-revision.md

```diff
@@ -0,0 +1,311 @@
+   safety.
+1) The `OnInvokeCallback` used by `CallbackAction` no longer takes a `FocusNode`
+   argument.
+1) The `ActionDispatcher.invokeAction` signature has changed to not accept an
+   optional `FocusNode`, but instead take an optional `BuildContext`.
+1) The `LocalKey` static constants (named key by convention) in `Action` subclasses
+   have been removed.
+1) The `Action.invoke` and `ActionDispatcher.invokeAction` methods now return the
+   result of invoking the action as an `Object`.
+1) The `Action` class may now be listened to for state changes.
+1) The `ActionFactory` typedef has been removed, as it is no longer used.
+
+## Example Analyzer Failures
+
+Here are some example analyzer failures that might be encountered where an
+outdated use of the Actions API might be the cause of the problem. The specifics
+of the error might differ, and there may be other failures caused by these
+changes.
+
+```
+error: MyActionDispatcher.invokeAction' ('bool Function(Action<Intent>, Intent, {FocusNode focusNode})') isn't a valid override of 'ActionDispatcher.invokeAction' ('Object Function(Action<Intent>, Intent, [BuildContext])'). (invalid_override at [main] lib/main.dart:74)
+
+error: MyAction.invoke' ('void Function(FocusNode, Intent)') isn't a valid override of 'Action.invoke' ('Object Function(Intent)'). (invalid_override at [main] lib/main.dart:231)
+
+error: The method 'isEnabled' isn't defined for the type 'Intent'. (undefined_method at [main] lib/main.dart:97)
+
+error: The argument type 'Null Function(FocusNode, Intent)' can't be assigned to the parameter type 'Object Function(Intent)'. (argument_type_not_assignable at [main] lib/main.dart:176)
+
+error: The getter 'key' isn't defined for the type 'NextFocusAction'. (undefined_getter at [main] lib/main.dart:294)
+
+error: The argument type 'Map<LocalKey, dynamic>' can't be assigned to the parameter type 'Map<Type, Action<Intent>>'. (argument_type_not_assignable at [main] lib/main.dart:418)
+```
+
+## Migration Guide
+
+Significant changes will be required to update existing code to the new API.
+
+### Actions mapping for pre-defined actions
+
+To update the action maps in the `Actions` widget for predefined actions in
+Flutter, like `ActivateAction` and `SelectAction`, etc., you'll need to:
+
+ - Update the argument type of the actions argument, and
+ - Use an instance of a specific Intent class in the Shortcuts mapping, rather than an Intent(TheAction.key) instance.
+
+Code before migration:
+```dart
+class MyWidget extends StatelessWidget {
+  // ...
+  @override
+  Widget build(BuildContext context) {
+    return Shortcuts(
+      shortcuts: <LogicalKeySet, Intent> {
+        LogicalKeySet(LogicalKeyboardKey.enter): Intent(ActivateAction.key),
+      },
+      child: Actions(
+        actions: <LocalKey, ActionFactory>{
+          Activate.key: () => ActivateAction(),
+        },
+        child: Container(),
+      )
+    );
+  }
+}
+```
+
+Code after migration:
```

Comment:
Add a <!--skip--> tag before this code. It is currently breaking the build, because our build tests against the stable release (1.12 at the moment).

Reply:
Done.  I added one before the block of analyzer warnings too, in case it's not looking at the "dart" part of the triple quotes.